pkg/scheduler/objects/application.go

```diff
@@ -973,6 +973,10 @@ func (sa *Application) tryAllocate(headRoom *resources.Resource, allowPreemption
 			// the iterator might not have the node we need as it could be reserved, or we have not added it yet
 			node := getNodeFn(requiredNode)
 			if node == nil {
+				if rateLimitedLog == nil {
```

Comment:
Instead of checking for null each time, write an accessor function for the log that uses sync.Once() to initialize it, and returns the initialized log.

Reply:
While we're at it, rename rateLimitedLog var to appLog.

Reply:
Done.